### PR TITLE
[ADD] customer balance on ticket

### PR DESCRIPTION
**Pull request by remytms**
## Description

This features depends on this PR: https://github.com/coopiteasy/addons/pull/235

I don't know how to automate tests for that.

## Odoo task (if applicable)

[task](https://gestion.coopiteasy.be/web#id=8573&model=project.task&view_type=form&menu_id=)

## Checklist before approval

- [ ] Tests are present (or not needed).
- [x] Credits/copyright have been changed correctly.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: account_customer_wallet/models/res_partner.py [hunk 1]
```diff
@@ -19,6 +19,9 @@ class Partner(models.Model):
         readonly=True,
         search="_search_customer_wallet_balance",
     )
+    is_customer_wallet_user = fields.Boolean(
+        string="Is Customer Wallet User",
+    )
```
Comment (carmenbianca) [inline account_customer_wallet/models/res_partner.py]: Should this not be a computed field? If there are one or more lines in the journal/account containing the (family of the) user, then this should be True, else False.

File: pos_customer_wallet/static/src/xml/pos.xml [hunk 2]
```diff
@@ -24,4 +24,45 @@
...
         </t>
     </t>
+
+    <t t-extend="PosTicket">
+        <!--
+            Customer must see his balance even if it does not use
+            customer wallet as a payement method for this order.
+            Because it's the only way he can check his balance.
+            Also this balance should not been shown if the user is not
+            allowed to use customer wallet as a payment method.
```
Comment (carmenbianca) [inline pos_customer_wallet/static/src/xml/pos.xml]: I think the word 'allowed' is misleading here. There's no logic (yet?) forbidding certain partners from using the customer wallet.

**Review by carmenbianca — COMMENTED** [review]
Good changes, a few comments.

**Comment by remytms** [comment]
@carmenbianca I agree with you, I'm in discussion with Pol to see if this `is_customer_wallet_user` is needed, because for me it brings only confusion and bug cases.

**Comment by polchampion** [comment]
@remytms @carmenbianca I agree that `is_customer_waller_user` as envisaged here isn't in keeping with the customer_wallet philosphy and poses practical problems - but the customers want a way to hide the total to non-users.

I anticipate the customers will want `is_customer_waller_user` to also hide the balance on the pos, and generally prevent non-users from using the wallet. A future development might also automatically set `is_customer_waller_user` to True when there is a movement on the customer wallet account for the corresponding partner. These improvements would come at a cost of complexity that seems too high compared to the benefit.

I see three options.
1. Keep this PR as is and see about future devs later.
2. Add `is_customer_waller_user` to `account_customer_wallet` BUT change the present PR to always display the balance on the ticket. Create an additional module that overcharges this behaviour and hides the ticket balance based on `is_customer_waller_user`.
3. DO NOT add `is_customer_waller_user` to `account_customer_wallet` and make a now module for it. Change `pos_customer_wallet` to always display the balance on the ticket. Use another module to hide the balance based on `is_customer_waller_user`.

What would you recommend?

**Comment by polchampion** [comment]
@remytms @carmenbianca I would value your input on this (see above) in order to discuss it with Rémy and decide what to do.

**Comment by carmenbianca** [comment]
@polchampion I think option 2 or 3 both work.

- Option 2 is good if `is_customer_wallet_user` is a computed field that simply checks whether the wallet was used at least once by the (family of the) user.
- Option 3 is good if it's a non-computed field.

**Comment by polchampion** [comment]
> @remytms @carmenbianca I agree that `is_customer_waller_user` as envisaged here isn't in keeping with the customer_wallet philosphy and poses practical problems - but the customers want a way to hide the total to non-users.
> 
> I anticipate the customers will want `is_customer_waller_user` to also hide the balance on the pos, and generally prevent non-users from using the wallet. A future development might also automatically set `is_customer_waller_user` to True when there is a movement on the customer wallet account for the corresponding partner. These improvements would come at a cost of complexity that seems too high compared to the benefit.
> 
> I see three options.
> 
>     1. Keep this PR as is and see about future devs later.
> 
>     2. Add `is_customer_waller_user` to `account_customer_wallet` BUT change the present PR to always display the balance on the ticket. Create an additional module that overcharges this behaviour and hides the ticket balance based on `is_customer_waller_user`.
> 
>     3. DO NOT add `is_customer_waller_user` to `account_customer_wallet` and make a now module for it. Change `pos_customer_wallet` to always display the balance on the ticket. Use another module to hide the balance based on `is_customer_waller_user`.
> 
> 
> What would you recommend?

As discussed with @remytms we go for option 3 and a non-computed field. If and when a future development is requested to 1) make this field computed and 2) prevent pos customers from using the wallet based on it, then we might move the field to account_customer_wallet.

**Comment by carmenbianca** [comment]
Rebased this PR to remove `is_customer_wallet_user` functionality. I will re-introduce it in new commits. If for some reason I don't get to that and someone else begins work on this; I still have the `is_customer_wallet_user` commit tagged locally.

**Comment by carmenbianca** [comment]
Rebased on top of 12.0.

**Review by carmenbianca — APPROVED** [review]

**Pull request merged**